### PR TITLE
18 create method to create jointnodes based on network

### PR DESCRIPTION
**Pull request by samvanderzwan**

### CODE REVIEW

File: src/simulator_core/entities/assets/junction.py [hunk 1]
```diff
@@ -35,7 +34,7 @@ def __init__(
         index: int = None,
     ):
         """Initialize a Junction object."""
-        self.pandapipes_net = pandapipes_net
```
Comment (vanmeerkerk) [inline src/simulator_core/entities/assets/junction.py:38]: Volgens mij kan je deze assignment niet weghalen zonder de _create methode aan te passen. Hoe kan je anders een netwerk opbouwen?

File: src/simulator_core/entities/esdl_object.py [hunk 2]
```diff
@@ -51,3 +53,30 @@ def get_all_assets_of_type(self, esdl_asset_type: str) -> list[EsdlAssetObject]:
...
+        This returns a list of list with the connected asset id and the port to which it is
+        connected to the asset.
+
+        :param str id: id of the asset for which we want to know the connected assets
+        :param Port port: port for which the connected assets need to be returned.
+        :return: List of list which the id of the connected assets and the connected port.
+        """
+        esdl_asset = self.energy_system_handler.get_by_id(id)
+        connected_assets = []
+        type_port = esdl.InPort
```
Comment (vanmeerkerk) [inline src/simulator_core/entities/esdl_object.py:69]: I would move this type_port statement to the if loop.

`
if port == Port.Out:
   type_port = esdl.OutPort
else:
   type_port = esdl.InPort
`

File: src/simulator_core/entities/esdl_object.py [hunk 3]
```diff
@@ -51,3 +53,30 @@ def get_all_assets_of_type(self, esdl_asset_type: str) -> list[EsdlAssetObject]:
...
+        if port == Port.Out:
+            type_port = esdl.OutPort
+        connected_port_ids = []
+        for esdl_port in esdl_asset.port:
+            if isinstance(esdl_port, type_port):
+                connected_port_ids = esdl_port.connectedTo
```
Comment (vanmeerkerk) [inline src/simulator_core/entities/esdl_object.py:75]: This seems odd. You're initializing a list and then overwriting the list with a single value. I believe you would like to append all connected ports to the "connected_port_ids" list. 

I also do not understand why you would like to break this statement. 

Can you combine the first loop with the second loop? You can directly assign the type of the connected asset. 

I would also make a separate method for the type check of the port. You are using it between line 69 and 71 and at this location.

File: unit_test/entities/test_entities.py [hunk 4]
```diff
@@ -24,9 +24,10 @@ class HeatNetworkTest(unittest.TestCase):
     def test_heat_network(self) -> None:
         """Generic/template test for Heatnetwork."""
         # Arrange
-
+        assets = []
+        junctions = []
         # Act
-        result = HeatNetwork()
+        result = HeatNetwork(assets, junctions)
```
Comment (vanmeerkerk) [inline unit_test/entities/test_entities.py:30]: What does the HeatNetwork(assets, junctions) call do when the assets and junctions are an empty list? Do we want to be able to create an empty HeatNetwork entity? 

If so, maybe we should add a default empty list already to the init of HeatNetwork? 

@MichielTukker

File: unit_test/entities/test_esdl_object.py [hunk 5]
```diff
@@ -37,14 +39,26 @@ def test_EsdlAssetObject(self):
         consumers = self.esdl_object.get_all_assets_of_type(consumer)
         pipes = self.esdl_object.get_all_assets_of_type(pipe)
         # Act
-        asset_producer = producers[0].convert_esdl()
```
Comment (vanmeerkerk) [inline unit_test/entities/test_esdl_object.py:40]: I think that this should be a separate commit? You are removing the convert_esdl method from the EsdlAssetObject? 

@MichielTukker

File: unit_test/transforms/test_mapper.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+class EsdlEnergySystemMapperTest(unittest.TestCase):
+
+    def test_to_entity(self):
+        # act
+        esdl_file_path = r'.\testdata\test1.esdl'
+        esdl_object = EsdlObject(pyesdl_from_file(esdl_file_path))
+        # arrange
+        network = EsdlEnergySystemMapper().to_entity(esdl_object)
+        # assert
+        self.assertTrue(True)
```
Comment (vanmeerkerk) [inline unit_test/transforms/test_mapper.py:31]: Why this assert statement?

Comment (vanmeerkerk) [reply unit_test/transforms/test_mapper.py:31]: I believe you want to check whether you have created a network asset?

**Review by vanmeerkerk — CHANGES_REQUESTED** [review]
Hey Sam,

Ik heb weer een aantal opmerkingen geplaatst. Zou je ook nog even naar de linting etc kunnen kijken?

File: src/simulator_core/entities/assets/junction.py (hunk 1 above)
Comment (samvanderzwan) [reply src/simulator_core/entities/assets/junction.py:38]: klopt, ik pas create method aan

File: src/simulator_core/entities/esdl_object.py (hunk 2 above)
Comment (samvanderzwan) [reply src/simulator_core/entities/esdl_object.py:69]: made a one line statement out of it.        
type_port = esdl.InPort if port == Port.Out else esdl.OutPort

File: src/simulator_core/entities/esdl_object.py (hunk 3 above)
Comment (samvanderzwan) [reply src/simulator_core/entities/esdl_object.py:75]: Not entirely if the asset does not have the requested type of port then the list would not exist and the next step would fail.

File: unit_test/entities/test_entities.py (hunk 4 above)
Comment (samvanderzwan) [reply unit_test/entities/test_entities.py:30]: If we implement more in heatnetwork class then this can also be covered. I already made an issue for extending the heat network class #42

File: unit_test/entities/test_esdl_object.py (hunk 5 above)
Comment (samvanderzwan) [reply unit_test/entities/test_esdl_object.py:40]: the convert method has been moved, so this test does not work anymore and the correct calls have been added

File: unit_test/transforms/test_mapper.py (hunk 6 above)
Comment (samvanderzwan) [reply unit_test/transforms/test_mapper.py]: Adjusted test

**Review by vanmeerkerk — APPROVED** [review]
Prima zo
